### ISSUE TITLE
Use bazel to generate test certificates

### ISSUE THREAD

**Issue by phlax** [envoyproxy/envoy#23912]
There have been a round of cert expiry that has had to be dealt with in CI recently

This has led to a lot of failed pipelines and the need to backport various fixes

One discussion around this has been to generate the certificates dynamically with bazel, and i guess some kind of `action_env` that expires/rebuilds at certain times

**Comment by ggreenway** [comment]
I'm strongly in favor of this. I don't know bazel well enough to do it myself.

**Comment by daixiang0** [comment]
Now they are generated by bash script, easily implement them in CI, just need one schedule job to update. I think no need to generate them every time in CI.

**Comment by phlax** [comment]
the first challenge is getting access to an openssl bin _in_ bazel - we could use the system bin but it would be better not to - something im realizing rebuilding the certs now - they require a very specific version

we certainly dont need to generate them every time in CI but im wondering if we need to somehow cause them to be rebuilt sometimes

**Comment by phlax** [comment]
seems like it should be pretty easy to get a boringssl bin https://github.com/google/boringssl/blob/master-with-bazel/BUILD#L181

**Comment by wbpcode** [comment]
considering the rebuilding is super-fast, I prefer the way just to generate it when we run the ci. Then any external robot/schedular is unnecessary for this.

**Comment by phlax** [comment]
i think we want to use bazel for all the reasons that we use bazel - which means i think that it will be built hermetically and cached - and not rebuilt unless its "inputs" change

not sure exactly how the certs are used in the tests or what expectations are - we could probably make the build have never expiring certs and not worry about needing to ever rebuild

(if the expiry was say 5 years then its hard to imagine a bazel cache lasting that long but it would break the hermetic model still)

**Comment by ggreenway** [comment]
I leave it up to the bazel-people whether 5 years in a cache could ever happen. 

I wonder if we could make an input to the step be a file that contains the `year:month` (even though the step doesn't use the file) to force it to refresh monthly.

**Comment by phlax** [comment]
yep that would be an `action_env` i think - altho it might also make other stuff expire too

relatedly im wondering if we can use similar to expire coverage/tests caches and resolve the ongoing timeout/server disappearring issues

**Comment by phlax** [comment]
one issue im realizing with this is that the current script breaks on ~recent versions of openssl (according to gpt >= 1.1.1), as you can no longer create back-dated expired certs - ie `-days -365`)

another reason that we should do this hermetically i think

cc @adisuissa wondering if you used a sufficiently old openssl when updating recently

**Comment by phlax** [comment]
nm - i see now - https://github.com/envoyproxy/envoy/pull/35781/files#diff-a00b822417222be53555c2e8f48b994b5e3f1e2d3962261167ba5afcee5d7883R321
